### ISSUE TITLE
Get rid of parameter to `peek`

### ISSUE THREAD

**Issue by ISSOtm** [gbdev/rgbds#708]
0 is the most common `peek` argument, and the grammar shouldn't require more than that.

Here is where `peek` is not called with 0 as its argument:
- [ ] [`peek`](https://github.com/gbdev/rgbds/blob/b3312886fb4b40ac4cb806d1a1410da35980c870/src/asm/lexer.c#L890) (recursively via `shiftChars`)
- [ ] [`shiftChars`](https://github.com/gbdev/rgbds/blob/b3312886fb4b40ac4cb806d1a1410da35980c870/src/asm/lexer.c#L931)
- [ ] [`yylex_NORMAL`](https://github.com/gbdev/rgbds/blob/b3312886fb4b40ac4cb806d1a1410da35980c870/src/asm/lexer.c#L1909) (`$ff00 + c`)
- [ ] [`yylex_NORMAL`](https://github.com/gbdev/rgbds/blob/b3312886fb4b40ac4cb806d1a1410da35980c870/src/asm/lexer.c#L1913) (via `shiftChars`)

The point of doing this would be to simplify expansion handling, so that `peek` could be implemented as shifting 1 char, and storing it in memory. This should be lighter than the current approach, notably by being more cache-friendly (I think?)

**Comment by Rangi42** [comment]
How about disallowing the spaces in `ld [$ff00+c], a`? The '`$ff00+c`' is meant to be a single token, and already has more limitations than a general-case `A + B` expression. For instance, you have to use hex '`$ff00`', not '`65280`' or '`($ff << 8)`'. Also, it doesn't handle line continuations or block comments; these are syntax errors:

```
    ld [$ff00 \
        + c], a

    ld [$ff00 /* start of HRAM */ + c], a
```

So '`ld [$ff00 + c], a`' may as well be an error too.

**Comment by aaaaaa123456789** [comment]
> How about just disallowing whitespace in `ld [$ff00+c], a`? The '`$ff00+c`' is meant to be a single token, and already has more limitations than a general-case `A + B` expression. For instance, you have to use hex '`$ff00`', not '`65280`' or '`($ff << 8)`'. Also, it doesn't handle line continuations or block comments; these are syntax errors:
> 
> ```
>     ld [$ff00 \
>         + c], a
> 
>     ld [$ff00 /* start of HRAM */ + c], a
> ```
> 
> So '`ld [$ff00 + c], a`' may as well be an error too.

The spacing was explicitly allowed at some point, mostly because people tend to put spaces around operators (and that `+` looks like one, even though it's not). It should be deprecated anyway (I've definitely brought up the 65280 issue before); `ldh [c], a` should be the preferred form. But what you say is how it used to work, and it was eventually changed on request.

**Comment by ISSOtm** [comment]
Closed by #799
